### PR TITLE
v1.1.0 - Use a search parameter to power return navigation

### LINKED ISSUE #3: Return capabilities are dropped when the browser refreshes from the target page and `location.state` cannot be used
Since the package utilizes the server provided `referrer` and that value is removed upon refresh, return navigation does not work when a page has return links, javascript is disabled, and the page is refreshed.

### PR DESCRIPTION
**Pull request by brian-dlee**
resolves #3

### CODE REVIEW

File: src/backward-link.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { useReturnLocation } from './hooks';
+
+export interface BackwardLinkProps extends Omit<LinkProps, 'to'> {
+  fallback?: To;
+}
+
+export function BackwardLink(props: BackwardLinkProps) {
+  const location = useReturnLocation() || props.fallback || '/';
+
+  return <Link to={location}>{props.children}</Link>;
```
Comment (zachrogo) [inline src/backward-link.tsx]: I think this should take a className prop. One thing we do a lot right now that we should stop doing is wrapping things like buttons in this case. That may have made some sense with a kit like MUI. But now that we have control over our css we should prefer instead to style our anchor tags as apposed to wrapping other styles/elements.

File: src/context.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  useEffect(() => {
+    const returnLocation = getReturnLocationFromSearch(
+      location.search,
+      props.defaultReturnLocationParam
+    );
+
+    if (returnLocation) {
+      setReturnLocation(returnLocation);
+    }
+  }, [location, navigate, props.defaultReturnLocationParam, props.referrer]);
```
Comment (zachrogo) [inline src/context.tsx]: Looks like you can kill navigate in this component

And if `props.defaultReturnLocationParam` and `props.referrer` are constantly changing, you can probs do more work outside the useeffect and keep the effect dependency real lean. It helps when craziness starts happening to easily reason about effects. 

```suggestion
    const returnLocation = getReturnLocationFromSearch(
      location.search,
      props.defaultReturnLocationParam
    );

  useEffect(() => {
    if (returnLocation) {
      setReturnLocation(returnLocation);
    }
  }, [returnLocation]);
```

File: src/index.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
-export * from './use-return-location'
-export * from './use-current-location'
-export * from './lib'
+export * from './backward-link';
+export * from './context';
+export * from './forward-link';
+export * from './hooks';
+export * from './lib';
```
Comment (zachrogo) [inline src/index.ts:5]: packager pattern ™️

**Review by zachrogo — APPROVED** [review]

File: src/index.ts (hunk 3 above)
Comment (brian-dlee) [reply src/index.ts:5]: Exactly! Gotta keep my ideology alive

File: src/context.tsx (hunk 2 above)
Comment (brian-dlee) [reply src/context.tsx]: The referrer should be generated only the loader, so it should never change the value and the default param is also meant to be a statically passed config option, so it shouldn't change either.

Comment (brian-dlee) [reply src/context.tsx]: And, I killed the navigate.

File: src/backward-link.tsx (hunk 1 above)
Comment (brian-dlee) [reply src/backward-link.tsx]: I forgot to pass all of `props` into the underlying component. I meant to just extend the component, but forgot to pas props in.

**Pull request merged**